committeeoversightapp/management/commands/import_data.py

```diff
@@ -99,82 +102,162 @@ def add_senate_committees(self):
+
+                    if event:
+                        self.update_hearing(event,
+                                            name,
+                                            start_date,
+                                            participating_committees,
+                                            classification,
+                                            category,
+                                            source,
+                                            source_hash)
                     else:
-                        participating_committees = self.get_participating_committees(committee1, committee2, subcommittee1, subcommittee2)
-                        event = self.match_by_date_and_participants(name, participating_committees, start_date, category, source, classification)
+                        self.create_hearing(name,
+                                            start_date,
+                                            participating_committees,
+                                            classification,
+                                            category,
+                                            source,
+                                            source_hash)
+
+            lugar_in_db = len(Event.objects.filter(sources__url=csvfile.name))
+
+            assert abs(lugar_in_db - i) < 5
+
+            with connection.cursor() as cursor:
+                    cursor.execute(
+                        '''DROP TABLE participants''')
 
-                        if event:
-                            self.update_hearing(event, category, source)
-                        else:
-                            self.create_hearing(name, start_date, participating_committees, classification, category, source)
 
-                self.row_count += 1
 
     def add_senate_hearings(self):
-        with open('data/final/senate.csv', 'r') as csvfile:
+        with open('data/final/senate.csv', 'r') as csvfile, \
+             transaction.atomic():
+
+            with connection.cursor() as cursor:
+                cursor.execute(
+                    '''CREATE TEMPORARY TABLE participants AS
+                       SELECT event_id,
+                              ARRAY_AGG(organization_id::text) AS committees
+                       FROM opencivicdata_eventparticipant
+                       GROUP BY event_id''')
+
             reader = csv.DictReader(csvfile)
-            self.row_count = 2
+            # we want to know the original row index for debugging purposes
+            reader = enumerate(reader, 2)
+            # ignore rows with a missing name
+            reader = ((i, row) for i, row in reader if row['Hearing/Report'])
 
-            for row in reader:
+            for i, (self.row_index, row) in enumerate(reader, 1):
                 name = row['Hearing/Report']
                 start_date = row['Date'].split('T', 1)[0]
-                source = row['source']
+                source = csvfile.name
```

Comment:
Why did you switch this to `csvfile.name`? This is saved in the url field of the source model, which I think isn't what we want on the front end: the site is expecting those to all be links when they exist. Maybe we could have different values for `note`, instead of making them all `spreadsheet` in this import.

Reply:
okay, I can fix this. there were some problems with the makefile and the source row was not correct for me,  so I didn't undrerstand this value.